tests/unit-tests/gconstruct/test_construct_graph.py

```diff
@@ -1274,6 +1277,407 @@ def test_parse_edge_data():
+            "name": "max_min_norm",
+            "max_val": 100,
+            "min_val": 0
+        }
+    }]
+    data = {
+        "test1": np.random.randint(0, 100, 3),
+        "test2": np.random.randint(0, 100, 3)
+    }
+    (res, _, _) = parse_feat_ops(feat_op5)
+    assert len(res) == 1
+    assert res[0].col_name == feat_op5[0]["feature_col"]
+    assert res[0].feat_name == feat_op5[0]["feature_name"]
+    assert isinstance(res[0], NumericalMinMaxTransform)
+    maxmin_feats = process_features(data, res, ext_mem=ext_mem)
+    assert "test3" in proc_res
+    assert proc_res["test3"].dtype == np.float32
+
+    data_maxmin1 = {
+        "test1": data["test1"]
+    }
+    feat_maxmin_single1 = [{
+        "feature_col": "test1",
+        "feature_name": "test3",
+        "transform":{
+            "name": "max_min_norm",
+            "max_val": 100,
+            "min_val": 0
+        }
+    }]
+    (res, _, _) = parse_feat_ops(feat_maxmin_single1)
+    maxmin_feat_single1 = process_features(data_maxmin1, res)
+
+    data_maxmin2 = {
+        "test2": data["test2"]
+    }
+    feat_maxmin_single2 = [{
+        "feature_col": "test2",
+        "feature_name": "test3",
+        "transform":{
+            "name": "max_min_norm",
+            "max_val": 100,
+            "min_val": 0
+        }
+    }]
+    (res, _, _) = parse_feat_ops(feat_maxmin_single2)
+    maxmin_feat_single2 = process_features(data_maxmin2, res)
+    maxmin_expec = np.column_stack((maxmin_feat_single1["test3"],
+                                maxmin_feat_single2["test3"]))
+    if isinstance(maxmin_feats, ExtMemArrayWrapper):
+        maxmin_feats = maxmin_feats.to_numpy()
+    assert_equal(maxmin_feats["test3"], maxmin_expec)
+
+    feat_op6 = [{
+        "feature_col": ["test1", "test2"],
+        "feature_name": "test3",
+        "transform":{
+            "name": "max_min_norm",
+        }
+    }]
+    try:
+        (res, _, _) = parse_feat_ops(feat_op6)
+        assert False, "expected Error raised for not " \
+                      "specifying max_val and min_val"
+    except AssertionError as e:
+        assert str(e) == "max_val and min_val for max_min_norm " \
+                         "feature transformation is needed"
+
+    feat_op7 = [{
+        "feature_col": ["test1", "test2"],
+        "feature_name": "test3",
+        "transform":{
+            "name": "to_categorical",
+        }
+    }]
+    try:
+        (res, _, _) = parse_feat_ops(feat_op7)
+        assert False, "expected Error raised for multi column on" \
+                      " categorical feature transformation"
+    except RuntimeError as e:
+        assert str(e) == "Do not support categorical feature " \
+                         "transformation on multiple columns"
+
+    feat_op8 = [{
+        "feature_col": [],
+        "feature_name": "test3"
+    }]
+    try:
+        (res, _, _) = parse_feat_ops(feat_op8)
+        assert False, "expected Error raised for invalid feature column"
```

Comment:
Please be more specific of what corner case you are testing. So other maintainer can understand what error you are testing here.